python_modules/dagster/dagster_tests/general_tests/test_pending_repository.py

```diff
@@ -160,11 +160,11 @@ def _op(context, res_upstream):
                 for cd in data
             ]
 
-    @asset(required_resource_keys={"foo"})
+    @asset
     def res_upstream(context):
         return context.resources.foo
 
-    @asset(required_resource_keys={"foo"})
+    @asset
```

Comment:
Tests using this were looking for `res_midstream` to be flagged specifically in the error message, when the same error would be thrown for all `res_upstream`, `res_downstream` as well. The reason the test did not previously fail is that `res_midstream` was incidentally being processed first, so that's where the error was. 

The adjustment to base asset job creation in this PR causes `res_midstream` to no longer be processed first (which does not matter), but broke this test. Therefore I removed the `required_resource_keys` on `res_upstream`/`res_downstream`, as this fixed the failing test and none of the other tests using these assets actually depends on `res_upstream`/`res_downstream` having these req'd keys.